### PR TITLE
add pr issue association rule

### PR DESCRIPTION
**Pull request by zhangyangyu**

### CODE REVIEW

File: src/contribute-to-tidb/contribute-code.md [hunk 1]
```diff
@@ -23,6 +23,12 @@ After a consensus is reached in issues, it's time to start the code contributing
 
 Clear and kind communication is key to this process.
 
+## Referring to an issue
+
+Code repos in TiDB community requires all the pull requests referring to corresponding issues. When opening a pull request, you MUST fill the pull request body with relevant issue numbers. In the pull request body, there MUST be one line matching the pattern `Issue Number: ref #{issue-number}` or `Issue Number: close[sd]?|resolve[sd]?|fix(e[sd])? #{issue-number}`. `ref` means associated with an issue but does not close it. There should be no other content in the same line and the check is case-insensitive. Multiple issues should be separated by a comma, like `Issue Number: close #12341, close #12342`. For PRs trying to close issues in a different repository, contributors need to first create an issue in the same repository and use this issue to track.
```
Comment (Mini256) [inline src/contribute-to-tidb/contribute-code.md:28]: I think direct and clear examples will be better understood than regular expressions.

As a guide, I think we should avoid long paragraphs of text, and let readers locate the part they need to read faster, by means of segments, tables, or code blocks.

<hr>

Code repositories in TiDB community require **ALL** the pull requests referring to its corresponding issues. In the pull request body, there **MUST** be one line starting with `Issue Number: ` and linking the relevant issues via the [keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword), for example:

If PR resolves the relevant issues, and you want GitHub to close these issues automatically after it merged into the default branch, you can use the syntax ( `KEYWORD #ISSUE-NUMBER`) like this:

```
Issue Number: close #123
```

If PR links with an issue but does not close it, you can use the keyword `ref` like this:

```
Issue Number: ref #456
```

Multiple issues should use full syntax for each issue and separate by a comma, like:

```
Issue Number: close #123, ref #456
```

For PRs trying to close issues in a different repository, contributors need to first create an issue in the same repository and use this issue to track.

Comment (zhangyangyu) [reply src/contribute-to-tidb/contribute-code.md:28]: Makes sense. I am happy to gradually change our dev guide to such style.

**Comment by zhangyangyu** [comment]
@all-contributors please add @Mini256  for content

**Pull request merged**